Dockerfile

```
@@ -1,100 +1,103 @@
+		$PHPIZE_DEPS \
+		libjpeg-turbo-dev \
+		libmcrypt-dev \
+		libmemcached-dev \
+		libpng-dev \
+		openldap-dev \
+		pcre-dev \
+		postgresql-dev \
+		sqlite-dev \
+	; \
+	\
+	docker-php-ext-configure gd --with-png-dir=/usr --with-jpeg-dir=/usr; \
+	docker-php-ext-install \
+		gd \
+		mysqli \
+		opcache \
+		pdo_mysql \
+		pdo_pgsql \
+		pdo_sqlite \
+		pgsql \
+		zip \
+	; \
+	\
+# pecl will claim success even if one install fails, so we need to perform each install separately
+	pecl install APCu-5.1.17; \
+	pecl install redis-4.3.0; \
+	\
+	docker-php-ext-enable \
+		apcu \
+		redis \
+	; \
+	\
+	runDeps="$( \
+		scanelf --needed --nobanner --format '%n#p' --recursive /usr/local/lib/php/extensions \
+			| tr ',' '\n' \
+			| sort -u \
+			| awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }' \
+	)"; \
+	apk add --virtual .postfixadmin-phpexts-rundeps $runDeps; \
+	apk del .build-deps
 
-RUN adduser -S -s /bin/bash -u 1001 -G root www-data
+# set recommended PHP.ini settings
+# see https://secure.php.net/manual/en/opcache.installation.php
+RUN { \
+		echo 'opcache.memory_consumption=128'; \
+		echo 'opcache.interned_strings_buffer=8'; \
+		echo 'opcache.max_accelerated_files=4000'; \
+		echo 'opcache.revalidate_freq=2'; \
+		echo 'opcache.fast_shutdown=1'; \
+	} > /usr/local/etc/php/conf.d/opcache-recommended.ini
 
-RUN touch /var/run/nginx.pid && \
-    chown -R www-data:root /var/run/nginx.pid /etc/php7/php-fpm.d
-
-RUN mkdir -p /var/www/html && \
-    mkdir -p /usr/share/nginx/cache && \
-    mkdir -p /var/cache/nginx && \
-    mkdir -p /var/lib/nginx && \
-    chown -R www-data:root /var/www /usr/share/nginx/cache /var/cache/nginx /var/lib/nginx/
+#VOLUME /var/www/html
+ENV COMPOSER_VERSION 1.9.0
 
 # Install composer
-RUN wget https://getcomposer.org/installer -O /tmp/composer-setup.php && \
-    wget https://composer.github.io/installer.sig -O /tmp/composer-setup.sig && \
-    php -r "if (hash('SHA384', file_get_contents('/tmp/composer-setup.php')) !== trim(file_get_contents('/tmp/composer-setup.sig'))) { unlink('/tmp/composer-setup.php'); echo 'Invalid installer' . PHP_EOL; exit(1); }" && \
-    php /tmp/composer-setup.php --version=$COMPOSER_VERSION --install-dir=bin && \
-    php -r "unlink('/tmp/composer-setup.php');"
+RUN set -eux; \
+	curl -fsSL https://getcomposer.org/installer -o /tmp/composer-setup.php; \
+	curl -fsSL https://composer.github.io/installer.sig -o /tmp/composer-setup.sig; \
+	php -r "if (hash('SHA384', file_get_contents('/tmp/composer-setup.php')) !== trim(file_get_contents('/tmp/composer-setup.sig'))) { unlink('/tmp/composer-setup.php'); echo 'Invalid installer' . PHP_EOL; exit(1); }"; \
+	php /tmp/composer-setup.php --version=$COMPOSER_VERSION --install-dir=/bin; \
+	php -r "unlink('/tmp/composer-setup.php');"
 
-WORKDIR /var/www/html/
-USER 1001
+ARG cachet_ver=2.3.18
+ARG archive_url=https://github.com/CachetHQ/Cachet/archive/v${cachet_ver}.tar.gz
```

```suggestion
ARG archive_url=https://github.com/CachetHQ/Cachet/archive/${cachet_ver}.tar.gz
```

related to convo in https://github.com/CachetHQ/Docker/pull/348#discussion_r405176641